pwndbg/commands/context.py

```diff
@@ -181,23 +199,57 @@ def context_code():
         if closest_line < 0:
             return []
 
-        source = gdb.execute('list %i' % closest_line, from_tty=False, to_string=True)
+        # Get the full source code
+        filename = symtab.fullname()
+        source = get_highlight_source(filename)
 
         # If it starts on line 1, it's not really using the
         # correct source code.
         if not source or closest_line <= 1:
             return []
 
-        # highlight the current code line
-        source_lines = source.splitlines()
-        if pwndbg.config.highlight_source:
-            for i in range(len(source_lines)):
-                if source_lines[i].startswith('%s\t' % closest_line):
-                    source_lines[i] = C.highlight(source_lines[i])
-                    break
-
-        banner = [pwndbg.ui.banner("source")]
-        banner.extend(source_lines)
+        n = int(source_code_lines)
+
+        # Compute the line range
+        start = max(closest_line - 1 - n//2, 0)
+        end = min(closest_line - 1 + n//2 + 1, len(source))
+        num_width = len(str(end))
+
+        # split the code
+        source = source[start:end]
+
+        # Compute the prefix_sign length
+        # XXX: use nearpc-prefix here, provide customize?
+        # XXX: copy from color/disasm
```

Comment:
What about those comments? We should probably delete them or change sth

Reply:
1 No idea. It depends on you, do you want people be able to customize the prefix sign for source code and disasm separately.

2. That means I think this part is a little dirty for calculating the string length. I mentioned that perhaps we can modified the setter of `Parameter` to force everything is unicode. But this should be in the other PR. For now, perhaps we can note this and put a `TODO` here?

Reply:
@cebrusfs yes please do, we have the same for nearpc and backtrace already.
```python
theme.Parameter('nearpc-prefix', '►', 'prefix marker for nearpc command')
theme.ColoredParameter('nearpc-prefix-color', 'none', 'color for nearpc command (prefix marker)')
```
would be neat to have that for code as well, ```code-prefix``` and ```code-prefix-color``` :heart:

Reply:
Added.